### ISSUE TITLE
Kasada Anti-Bot

### ISSUE THREAD

**Issue by 0x6a69616e** [ading2210/vercel-llm-api#22]
During a reanalysis of Vercel AI's functionality, I came across two additional headers used in the HTTP requests of its generational endpoints, namely `x-kpsdk-cd`, `x-kpsdk-ct`, `x-kpsdk-im`. After some research, I found that these headers are related to [Kasada](https://www.kasada.io/).

I've already done a bit of reverse engineering. Achieving the value of `x-kpsdk-ct` and `x-kpsdk-im` was relatively easy, but it seems `x-kpsdk-cd` is challenging to figure out.

If the headers' values aren't in any way valid, the API will respond with a 429 error.

**Comment by viemccoy** [comment]
Yeah my previous utilization of this tool has been broken recently, I assume by this. Even using the retry until a response trick doesn't work anymore. Bummer because this is such a great way to test different models!

**Comment by sxqib** [comment]
I found something: https://github.com/Pr0t0ns/Kasada-Reverse

**Comment by 0x6a69616e** [comment]
Streamlink's Twitch plugin leverages CDP and simple script injection to bypass Kasada protections on Twitch. I suggest you have a look at the source code to see if you can get anything from it.

[https://github.com/streamlink/streamlink/blob/58acbccccb03c920cc47ce17025a3ebb4bf25fb5/src/streamlink/plugins/twitch.py#L490](https://github.com/streamlink/streamlink/blob/58acbccccb03c920cc47ce17025a3ebb4bf25fb5/src/streamlink/plugins/twitch.py#L490)

**Comment by 0x6a69616e** [comment]
I've actually developed a Kasada solver myself, taking a bit of inspiration from their bypass concept.

[0x6a69616e/kpsdk-solver](https://github.com/0x6a69616e/kpsdk-solver)

But kpsdk-solver is entirely written in JavaScript, which makes this pretty much a no-go.

**Comment by metimol** [comment]
Maybe in gpt4free working?

https://github.com/xtekky/gpt4free/blob/main/g4f%2FProvider%2FVercel.py
